src/Storages/MergeTree/KeyCondition.cpp

```diff
@@ -3171,15 +3171,9 @@ BoolMask KeyCondition::checkInHyperrectangle(
             /// Close ring
             boost::geometry::correct(polygon_by_minmax_index);
 
-            using MultiPolygon = boost::geometry::model::multi_polygon<RPNElement::Polygon>;
-            MultiPolygon intersection{};
-            boost::geometry::intersection(polygon_by_minmax_index, element.polygon, intersection);
-            has_intersection = !intersection.empty();
-
-            if (has_intersection)
-                rpn_stack.emplace_back(true, true);
-            else
-                rpn_stack.emplace_back(false, true);
+            rpn_stack.emplace_back(
```

Comment:
hi alixey, maybe we should keep the old logic, because the polygon may have a hole the the `can_be_false` should be true.

Reply:
Yes, you are right, let's keep "can_be_false" as true.